From 7ae4f01cdfc48ed8cbc87030bf272d52fd695845 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Fri, 28 Sep 2018 11:05:06 +0530
Subject: [PATCH 0431/1851] v4l: xilinx: vpss-scaler: Fix array out of bound
 access

commit 0040df6b111ce204701726cc10654c22145c7199 from
https://github.com/Xilinx/linux-xlnx.git

This patch fixes random kernel crashes caused while accessing and
modifying the vertical scaler coefficients.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-vpss-scaler.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-vpss-scaler.c b/drivers/media/platform/xilinx/xilinx-vpss-scaler.c
index 394d11343d65..21ec4d293278 100644
--- a/drivers/media/platform/xilinx/xilinx-vpss-scaler.c
+++ b/drivers/media/platform/xilinx/xilinx-vpss-scaler.c
@@ -1002,10 +1002,11 @@ xv_vscaler_load_ext_coeff(struct xscaler_device *xscaler,
 			/* pad left */
 			for (j = 0; j < offset; j++)
 				xscaler->vscaler_coeff[i][j] = 0;
+			/* pad right */
+			j = ntaps + offset;
+			for (; j < XV_VSCALER_MAX_V_TAPS; j++)
+				xscaler->vscaler_coeff[i][j] = 0;
 		}
-		/* pad right */
-		for (j = (ntaps + offset); j < XV_VSCALER_MAX_V_TAPS; j++)
-			xscaler->vscaler_coeff[i][j] = 0;
 	}
 }
 
-- 
2.31.1

